lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -47,11 +52,36 @@ func (c *WorkloadEndpointClient) Create(ctx context.Context, kvp *model.KVPair)
 }
 
 func (c *WorkloadEndpointClient) Update(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
-	log.Warn("Operation Update is not supported on WorkloadEndpoint type")
-	return nil, cerrors.ErrorOperationNotSupported{
-		Identifier: kvp.Key,
-		Operation:  "Create",
+	log.Debug("Received Update request on WorkloadEndpoint type")
+
+	// We only patch the existing Pod to include an IP address, if one has been
+	// set on the workload endpoint.
+	// TODO: This is only required as a workaround for an upstream k8s issue.  Once fixed,
+	// this should be a no-op. See https://github.com/kubernetes/kubernetes/issues/39113
+	ips := kvp.Value.(*apiv2.WorkloadEndpoint).Spec.IPNetworks
+	if len(ips) > 0 {
+		log.Debugf("Applying workload with IPs: %+v", ips)
+		weid, err := c.converter.ParseWorkloadName(kvp.Key.(model.ResourceKey).Name)
+		if err != nil {
+			return nil, err
+		}
+		if weid.Pod == "" {
+			return nil, cerrors.ErrorInsufficientIdentifiers{Name: kvp.Key.(model.ResourceKey).Name}
+		}
+		ns := kvp.Key.(model.ResourceKey).Namespace
+		pod, err := c.clientSet.CoreV1().Pods(ns).Get(weid.Pod, metav1.GetOptions{})
+		if err != nil {
+			return nil, K8sErrorToCalico(err, kvp.Key)
+		}
+		pod.Status.PodIP = ips[0]
+		pod, err = c.clientSet.CoreV1().Pods(ns).UpdateStatus(pod)
```

Comment:
We can let this slide for now, but this really needs to be updated to use `Patch` - right now we're going to be overwriting other people's status updates if there are any.  There's already an issue to track, but just noting here to raise awareness!

Reply:
Regarding not parsing a pod with no Nodename - To ensure this behavior, I'll  check if such a FV test exist and if not add one (to the soon to be restored test file).